client/reader/following-edit/subscribe-form.jsx

```diff
@@ -98,11 +99,17 @@ var FollowingEditSubscribeForm = React.createClass( {
 	},
 
 	isWellFormedFeedUrl: function( parsedUrl ) {
-		if ( parsedUrl.hostname && parsedUrl.hostname.indexOf( '.' ) !== -1 ) {
-			return true;
+		if ( ! parsedUrl.hostname || parsedUrl.hostname.indexOf( '.' ) === -1 ) {
+			return false;
 		}
 
-		return false;
+		// Check for a valid-looking TLD
+		const lastHostnameSegment = last( parsedUrl.hostname.split( '.' ) );
```

Comment:
i think this is fine, but you could also use `lastIndexOf`, compare that to length - 2 (3?), and avoid the allocations that come with split.